engine/src/main/java/org/terasology/rendering/world/WorldRendererImpl.java

```diff
@@ -249,15 +252,10 @@ private void initRenderGraph() {
         renderGraph.add(bloomPassesNode);
         renderGraph.add(blurPassesNode);
         renderGraph.add(finalPostProcessingNode);
-    }
 
-    private static <T extends Node> T createInstance(Class<T> type, Context context) {
-        // Attempt constructor-based injection first
-        T node = InjectionHelper.createWithConstructorInjection(type, context);
-        // Then fill @In fields
-        InjectionHelper.inject(node, context);
-        node.initialise();
-        return type.cast(node);
+        RenderPipelineGenerator generator = new RenderPipelineGenerator(renderGraph);
+        RenderPipelineOptimizer optimizer = new RenderPipelineOptimizer(generator.generate());
+        renderPipelineExecutor = new RenderPipelineExecutor(optimizer.optimize());
```

Comment:
After reading your comment and _by instinct_ now I think having `createFrom(listofNodes)` makes more sense. Since it's not a big change, whenever we feel like the other way around is better, it wouldn't be a problem to modify it.